### PR TITLE
API-1509: Enable aes-gcm encryption provider

### PR DESCRIPTION
**Pull request by dgrisonnet**
proof PR: https://github.com/openshift/cluster-kube-apiserver-operator/pull/1446

### CODE REVIEW

**Comment by dgrisonnet** [comment]
The e2e-aws-encryption seem to have been broken for a while: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_library-go/1469/pull-ci-openshift-library-go-master-e2e-aws-encryption/1626266910360342528/artifacts/e2e-aws-encryption/test/build-log.txt. Although the job succeeds, it doesn't do anything at the moment.

**Review by dinhxuanvu — COMMENTED** [review]
Looking good

File: pkg/operator/encryption/crypto/keys.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 var (
 	ModeToNewKeyFunc = map[state.Mode]func() []byte{
 		state.AESCBC:    NewAES256Key,
+		state.AESGCM:    NewAES256Key,
```
Comment (deads2k) [inline pkg/operator/encryption/crypto/keys.go:12]: I didn't realize it was exactly the same key. That's convenient.

**Comment by dgrisonnet** [comment]
@deads2k can you please have another look at this PR now that the API has been merged?

**Pull request merged**